junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/InvocationAwareExecutableInvoker.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.execution;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+import static org.junit.jupiter.engine.execution.ParameterResolutionUtils.resolveParameters;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+import org.apiguardian.api.API;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.InvocationInterceptor.Invocation;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
+import org.junit.jupiter.engine.extension.ExtensionRegistry;
+
+/**
+ * {@code InvocationAwareExecutableInvoker} encapsulates the invocation of a
+ * {@link java.lang.reflect.Executable} (i.e., method or constructor),
+ * including support for dynamic resolution of method parameters via
+ * {@link ParameterResolver ParameterResolvers}.
+ *
+ * @since 5.0
+ */
+@API(status = INTERNAL, since = "5.0")
+public class InvocationAwareExecutableInvoker {
```

Comment:
I'm wondering if including `InvocationInterceptor` in the name would be better than `InvocationAware`.

So maybe something like `InvocationInterceptorExecutableInvoker`, `InvocationInterceptingExecutableInvoker`, or `InterceptingExecutableInvoker`?

I know: naming is hard. 😉 

@marcphilipp, did you have another idea for this name?

Reply:
I'd prefer `Intercepting` rather than `Interceptor`, since it qualifies the word "invoker"

Reply:
So.... `InterceptingExecutableInvoker`?